exp/controllers/awsmachinepool_controller.go

```diff
@@ -404,9 +426,16 @@ func (r *AWSMachinePoolReconciler) updatePool(machinePoolScope *scope.MachinePoo
 		"subnets of machinePoolScope", subnetIDs,
 		"subnets of existing asg", existingASG.Subnets)
 	less := func(a, b string) bool { return a < b }
-	subnetChanges := cmp.Diff(subnetIDs, existingASG.Subnets, cmpopts.SortSlices(less)) != ""
+	subnetChanges := cmp.Diff(subnetIDs, existingASG.Subnets, cmpopts.SortSlices(less))
```

Comment:
nit: Let's please use a consistent suffix in identifiers that represent "diffs."

I humbly suggest `subnetsDiff` (note the plural) and `asgDiff`.

Reply:
sgtm